src/xmpp/message.c

```diff
@@ -258,6 +258,11 @@ _message_handler(xmpp_conn_t* const conn, xmpp_stanza_t* const stanza, void* con
             free(mybarejid);
         }
 
+        // Prevent duplicate messages when current client is sending a message
+        if (msg_stanza && g_strcmp0(xmpp_stanza_get_from(msg_stanza), xmpp_conn_get_jid(conn)) == 0) {
```

Comment:
This will prevent all further parsing of the stanza. Including `_handle_chat_states`